src/components/edit-details/index.js

```diff
@@ -16,12 +16,27 @@ import {
 import './style.css';
 export default class EditDetails extends Component {
   onSearchClick () {
+    const { go } = this.props;
     dataLayer.push({
       event: 'travelInfoUpdate'
     });
     this.props.updateHeaderTitles();
     this.props.startSearch();
-    this.props.goBack();
+    // if statement checks if the browser is safari
+    // using go(-2) for safari instead of go(-1) because it hangs
+    if (Object.prototype.toString.call(window.HTMLElement).indexOf('Constructor') > 0) {
```

Comment:
definitely add a _link_ to where you discovered that this is a good way to browser-dedect for Safari...